src/persist/src/indexed/cache.rs

```diff
@@ -130,22 +127,19 @@ impl<B: BlobRead> BlobCache<B> {
     /// thread if it's not already in the cache.
     pub fn get_unsealed_batch_async(&self, key: &str) -> PFuture<Arc<BlobUnsealedBatch>> {
         let (tx, rx) = PFuture::new();
-        {
-            // New scope to ensure the cache lock is dropped during the
-            // (expensive) get.
-            let unsealed = match self.unsealed.lock() {
-                Ok(unsealed) => unsealed,
-                Err(err) => {
-                    tx.fill(Err(err.into()));
-                    return rx;
-                }
-            };
-            if let Some(entry) = unsealed.get(key) {
+        match self.cache.get_unsealed(key) {
+            Err(err) => {
+                tx.fill(Err(err));
```

Comment:
it's a little weird to fill with an error given that we could just fetch it, but OTOH I don't want to drop the error on the floor. I think what you have is probably the right thing for now, but something to mull over

Reply:
added a todo that we should revisit. this is also what we were doing previously

Reply:
oh we sure are. i missed that when i left this comment. a TODO seems worthwhile anyway